src/users_db.rs

```diff
@@ -150,9 +163,45 @@ impl UsersDb {
         )
     }
 
-    pub fn read(&self) -> rusqlite::Result<Vec<User>> {
-        let mut stmt = try!(self.connection.prepare("SELECT * FROM users"));
-        let rows = try!(stmt.query(&[]));
+    pub fn read(&self, filter: ReadFilter) -> rusqlite::Result<Vec<User>> {
+        let mut stmt = try!(
+            self.connection.prepare("SELECT * FROM users")
```

Comment:
If we want to do the prepare for all the different matches of `filter` should this one be under the filter All?